iree/compiler/Conversion/LinalgToLLVM/ConvertToLLVM.cpp

```diff
@@ -217,22 +226,34 @@ class ConvertFuncWithHALInterface : public ConvertToLLVMPattern {
     }
 
     TypeConverter::SignatureConversion signatureConverter(/*numOrigInputs=*/0);
-
-    // func foo(%packed_buffer_args: !llvm<i8**>, %push_constant: !llvm<i32*>)
+    // clang-format off
+    // func foo(%packed_buffer_args: !llvm<i8**>, %push_constant: !llvm<i32*>, thread_idx: i32, thread_idy, thread_idz: i32)
+    // clang-format on
     MLIRContext *context = rewriter.getContext();
     auto packedBuffersArgsTy =
         LLVM::LLVMType::getInt8PtrTy(context).getPointerTo();
     auto pushConstantArgTy = LLVM::LLVMType::getInt32Ty(context).getPointerTo();
+    auto threadIdXTy = LLVM::LLVMType::getInt32Ty(context);
+    auto threadIdYTy = LLVM::LLVMType::getInt32Ty(context);
+    auto threadIdZTy = LLVM::LLVMType::getInt32Ty(context);
     signatureConverter.addInputs(packedBuffersArgsTy);
     signatureConverter.addInputs(pushConstantArgTy);
+    signatureConverter.addInputs(threadIdXTy);
+    signatureConverter.addInputs(threadIdYTy);
+    signatureConverter.addInputs(threadIdZTy);
 
     // Create the new function's signature.
     Location loc = funcOp.getLoc();
     auto newFuncOp = rewriter.create<FuncOp>(
         loc, funcOp.getName(),
         rewriter.getFunctionType(signatureConverter.getConvertedTypes(),
-                                 llvm::None),
-        ArrayRef<NamedAttribute>());
+                                 llvm::None));
+
+    // TODO(ataei): Constructor above should take all attributes.
+    auto nmWorkgroupsFnAttrName = funcOp.getAttr(getNumWorkgroupsFnAttrName());
+    if (nmWorkgroupsFnAttrName) {
```

Comment:
I am confused whats happening here. You are checking if the attribute exists, and if it exists setting the same attribute back?

Reply:
Fixed now =)